File: app/src/main/java/com/example/tree/zhihu/activity/SplashActivity.java
package com.example.tree.zhihu.activity;

import android.content.Intent;
import android.os.Handler;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.TextView;

import com.example.tree.zhihu.R;

public class SplashActivity extends AppCompatActivity {

    TextView splash_1;
    TextView splash_2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
        splash_1 = (TextView) findViewById(R.id.splash_1);
        splash_2 = (TextView) findViewById(R.id.splash_2);
        Thread myThread=new Thread(){//创建子线程
            @Override
            public void run() {
                try{
                    sleep(3000);//使程序休眠
                    Intent it=new Intent(getApplicationContext(), MainActivity.class);//启动MainActivity
                    startActivity(it);
                    finish();//关闭当前活动
                }catch (Exception e){
                    e.printStackTrace();
                }
            }
        };
        myThread.start();//启动线程
    }
}

File: app/src/main/java/com/example/tree/zhihu/activity/SplashActivity.kt
package com.example.tree.zhihu.activity

import android.content.Intent
import android.os.Handler
import android.os.Message
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.TextView

import com.example.tree.zhihu.R

class SplashActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
        val myThread = object : Thread() {
            //创建子线程
            override fun run() {
                try {
                    Thread.sleep(3000)//使程序休眠
                    val it = Intent(applicationContext, MainActivity::class.java)//启动MainActivity
                    startActivity(it)
                    finish()//关闭当前活动
                } catch (e: Exception) {
                    e.printStackTrace()
                }

            }
        }
        myThread.start()//启动线程
    }
}
